lib/drives/models/drive.dart

```diff
@@ -57,4 +59,25 @@ class Drive extends Model {
   String toString() {
     return 'Drive{id: $id, from: $start at $startTime, to: $end at $endTime, by: $driverId}';
   }
+
+  static Future<Drive?> userAlreadyHasDrive(
+      DateTime start, DateTime end, int userId) async {
+    final List<dynamic> drivesList =
+        await supabaseClient.from('drives').select().eq('driver_id', userId);
+    final List<Drive> drives = drivesList
```

Comment:
there is a `fromJsonList` method you can use

Reply:
i didn't understand how because he wouldn't cast the List<dynamic> to List<Map<String, dynamic>>

Reply:
are you sure? the compiler does not show problems for me
```suggestion
    final List<Map<String, dynamic>> drivesList =
        await supabaseClient.from('drives').select().eq('driver_id', userId);

    final List<Drive> drives = Drive.fromJsonList(drivesList);
```

Reply:
yeah i know but if you run it, an exception is thrown:
_E/flutter (30537): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: type 'List<dynamic>' is not a subtype of type 'List<Map<String, dynamic>>'_

Reply:
okay that's weird... but if it does not work anyways, then let's just change the signature of the JsonList method

Reply:
yes i thought about that too.

Reply:
changed it.